### PR TITLE
add some pmap tests

### PR DESCRIPTION
**Pull request by mattjj**
Checked on single-CPU (1 replica) and TPU (2 replicas). On GPU some tests run into the issue of the ReplicaId HLO not being implemented on GPU, which we can work around (but this PR doesn't yet).

### CODE REVIEW

File: jax/interpreters/pxla.py [hunk 1]
```diff
@@ -109,6 +109,21 @@ def assign_shards_to_replicas(nrep, size):
   return tuple(indices.ravel())
 
 def replica_groups(nrep, mesh_spec, mesh_axis):
+  """Compute XLA replica groups from a replica count and device mesh data.
+
+  ArgS:
```
Comment (hawkinsp) [inline jax/interpreters/pxla.py]: Args (typo)

File: jax/interpreters/pxla.py [hunk 2]
```diff
@@ -109,6 +109,21 @@ def assign_shards_to_replicas(nrep, size):
...
+    nrep: int, number of relpicas (a computation-dependent value).
```
Comment (hawkinsp) [inline jax/interpreters/pxla.py]: replicas (typo)

**Review by hawkinsp — COMMENTED** [review]
Looks pretty good to me.

File: jax/interpreters/pxla.py (hunk 1 above)
Comment (mattjj) [reply jax/interpreters/pxla.py]: Done.

File: jax/interpreters/pxla.py (hunk 2 above)
Comment (mattjj) [reply jax/interpreters/pxla.py]: Done.

**Pull request merged**